File: backend/app/schemas/booking.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class BookingCreate(BaseModel):
    user_name: str
    user_phone: str
    offer_id: int
    booked_at: datetime
    quantity: int = 1
    total_price: float

class BookingResponse(BaseModel):
    id: int
    booking_code: str
    user_name: str
    user_phone: str
    offer_id: int
    booked_at: datetime
    quantity: int
    total_price: float
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

File: syria_offers_app/lib/models/booking.dart
class Booking {
  final String userName;
  final String userPhone;
  final DateTime bookedAt;
  final int quantity;
  final double totalPrice;
  final int offerId;

  Booking({
    required this.userName,
    required this.userPhone,
    required this.bookedAt,
    required this.quantity,
    required this.totalPrice,
    required this.offerId,
  });
}